transformations/writing_system_replacement/README.md

```diff
@@ -0,0 +1,116 @@
+
+|writing system type | some real-world languages that use it | "I love potatoes" (a sample output in a generated writing system)      |
+|---------------  | ------------------------------------  | -------------------------------------- |                                
+|alphabet         | English, Spanish, Russian             | 之 笓䒉㘔䆇 躓䒉蝲討蝲䒉䆇䁣             |
+|syllabary        | Mycenaean Greek, Japanese             | 䃓 熵鏘 㚐料                           |
+|partial phonemic | Hebrew,  Arabic                       | 阠㚶乍 渓绌敿                          |
+|logographic      | Ancient Egyptian, Oracle bone script  | 驿掩㑇㕶誨                             |
+
+## What tasks does it intend to benefit?
+This transformation could benefit text classification tasks (especially - language identification tasks).
+
+Trained humans are able to read texts that are written in different writing systems.
+For example, Japanese speakers can easily identify these texts as Japanese, and understand their meaning 
+(all 3 means "hiragana"):
+
+    平仮名
+    ひらがな
+    ヒラガナ
+
+The ability of ML models to identify and understand a language written in various writing systems could
+improve the quality of language identification in general, 
+and the quality of identification of low-resource languages in particular.
+
+As a useful side effect, it could help to decipher the [Voynich manuscript](https://en.wikipedia.org/wiki/Voynich_manuscript),
+the [Phaistos Disc](https://en.wikipedia.org/wiki/Phaistos_Disc), and other undeciphered artifacts.
+
+## Robustness Evaluation
+
+We've evaluated this transformation by running the build-in evaluate.py script as follows:
+`python3 evaluate.py -t WritingSystemReplacement`
+
+Note: it seems that there is a bug in 
+[this line](https://github.com/GEM-benchmark/NL-Augmenter/blob/f0111c1587cfa36cd4bd2c9739744e59c2796c26/TestRunner.py#L132)
+in TestRunner.py, making the script crash during any test, 
+which we temporally fixed by replacing the line's right side with `str(package_dir.parent.joinpath(search))`
+
+To speed up the test, we also removed all other transformations from the local NL-Augmenter/transformations dir.
+
+The results of the single run are as follows:
+
+```
+Here is the performance of the model aychang/roberta-base-imdb on the test[:20%] split of the imdb dataset
+The accuracy on this subset which has 1000 examples = 96.0
+Applying transformation:
+100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1000/1000 [00:19<00:00, 51.83it/s]
+Finished transformation! 1000 examples generated from 1000 original examples, with 1000 successfully transformed and 0 unchanged (1.0 perturb rate)
+Here is the performance of the model on the transformed set
+The accuracy on this subset which has 1000 examples = 100.0
+
+```
+
+The performance is surprisingly good, considering the dramatic changes made by our transformation.
```

Comment:
Consider adding the 4 suggested tests for Text Classification in the [evaluation](https://github.com/GEM-benchmark/NL-Augmenter/tree/main/evaluation)  section. You will notice that in the 3 non-default tests the performance falls significantly, as expected:

- `roberta-base-SST-2`: 94.0 -> 51.0
- `bert-base-uncased-QQP`: 92.0 -> 67.0
- `roberta-large-mnli`: 91.0 -> 43.0

It is still very interesting how the performance improves to 100.0 for `roberta-base-imdb`. It might be worth looking into that.

Reply:
Thank you for the performance evaluation! Added the results to the readme.

Regarding the imdb anomaly, I'll open an issue about it.